lib/ansible/modules/cloud/dimensiondata/dimensiondata_vlan.py

```diff
@@ -0,0 +1,571 @@
+        private_ipv6_base_address:
+            description: The base address for the VLAN's IPV6 network.
+            type: string
+            sample: 2402:9900:111:1195:0:0:0:0
+        private_ipv6_prefix_size:
+            description: The prefix size for the VLAN's IPV6 network.
+            type: int
+            sample: 64
+        private_ipv6_gateway_address:
+            description: The gateway address for the VLAN's IPV6 network.
+            type: string
+            sample: 2402:9900:111:1195:0:0:0:1
+        status:
+            description: VLAN status.
+            type: string
+            sample: NORMAL
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.dimensiondata import DimensionDataModule, UnknownNetworkError
+
+try:
+    from libcloud.common.dimensiondata import DimensionDataVlan, DimensionDataAPIException
+
+    HAS_LIBCLOUD = True
+
+except ImportError:
+    DimensionDataVlan = None
+
+    HAS_LIBCLOUD = False
+
+
+class DimensionDataVlanModule(DimensionDataModule):
+    """
+    The dimensiondata_vlan module for Ansible.
+    """
+
+    def __init__(self):
+        """
+        Create a new Dimension Data VLAN module.
+        """
+
+        super(DimensionDataVlanModule, self).__init__(
+            module=AnsibleModule(
+                argument_spec=DimensionDataModule.argument_spec_with_wait(
+                    name=dict(required=True, type='str'),
+                    description=dict(default='', type='str'),
+                    network_domain=dict(required=True, type='str'),
+                    private_ipv4_base_address=dict(default='', type='str'),
+                    private_ipv4_prefix_size=dict(default=0, type='int'),
+                    allow_expand=dict(required=False, default=False, type='bool'),
+                    state=dict(default='present', choices=['present', 'absent', 'readonly'])
+                ),
+                required_together=DimensionDataModule.required_together()
+            )
+        )
+
+        self.name = self.module.params['name']
+        self.description = self.module.params['description']
+        self.network_domain_selector = self.module.params['network_domain']
+        self.private_ipv4_base_address = self.module.params['private_ipv4_base_address']
+        self.private_ipv4_prefix_size = self.module.params['private_ipv4_prefix_size']
+        self.state = self.module.params['state']
+        self.allow_expand = self.module.params['allow_expand']
+
+        if self.wait and self.state != 'present':
+            self.module.fail_json(
+                msg='The wait parameter is only supported when state is "present".'
+            )
+
+    def state_present(self):
+        """
+        Ensure that the target VLAN is present.
+        """
+
+        network_domain = self._get_network_domain()
+
+        vlan = self._get_vlan(network_domain)
+        if not vlan:
+            if self.module.check_mode:
```

Comment:
I don't think this nets the functionality of `check_mode`, the idea is to return `changed: True` if a change would have otherwise taken place if the task was run not in check mode.